packages/core/admin/admin/src/content-manager/components/BlocksInput/Blocks/Code.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+
+import { Code } from '@strapi/icons';
+import { Transforms } from 'slate';
+import styled from 'styled-components';
+
+import { type BlocksStore } from '../BlocksEditor';
+
+const CodeBlock = styled.pre.attrs({ role: 'code' })`
+  border-radius: ${({ theme }) => theme.borderRadius};
+  background-color: ${({ theme }) => theme.colors.neutral100};
+  max-width: 100%;
+  overflow: auto;
+  padding: ${({ theme }) => `${theme.spaces[3]} ${theme.spaces[4]}`};
+  flex-shrink: 0;
+  & > code {
+    font-family: 'SF Mono', SFMono-Regular, ui-monospace, 'DejaVu Sans Mono', Menlo, Consolas,
+      monospace;
+    color: ${({ theme }) => theme.colors.neutral800};
+    overflow: auto;
+    max-width: 100%;
+  }
+`;
+
+const codeBlocks: Pick<BlocksStore, 'code'> = {
```

Comment:
do you think satisfies would be more appropriate? https://www.totaltypescript.com/clarifying-the-satisfies-operator

Reply:
love this, thanks!

Reply:
I think it can be very useful in the future if we allow custom blocks. But I'm not sure it can be applied here though

>When you use a colon, the type BEATS the value.

I think in my case my type describes exactly the value I want, no key could be added or removed, so I'm happy with the type beating the value here